src/Storages/PartitionedSink.cpp

```diff
@@ -23,6 +26,87 @@ namespace ErrorCodes
     extern const int CANNOT_PARSE_TEXT;
 }
 
+namespace Setting
+{
+extern const SettingsBool use_hive_partitioning;
```

Comment:
This setting already exists, and is currently used to create virtual columns with partition key/values extracted from filepath in order to apply filters without having to read the actual files.

Perhaps a new engine setting should be introduced to control this behavior?